src/launchpad/kafka.py

```diff
@@ -16,23 +18,52 @@
 from arroyo.processing.strategies import ProcessingStrategy, ProcessingStrategyFactory
 from arroyo.processing.strategies.commit import CommitOffsets
 from arroyo.processing.strategies.healthcheck import Healthcheck
-from arroyo.processing.strategies.run_task_in_threads import RunTaskInThreads
+from arroyo.processing.strategies.run_task_with_multiprocessing import (
+    ChildProcessTerminated,
+    MultiprocessingPool,
+    RunTaskWithMultiprocessing,
+)
 from arroyo.types import Commit, Partition
 from sentry_kafka_schemas import get_codec
 
+from launchpad.artifact_processor import ArtifactProcessor
 from launchpad.constants import (
     HEALTHCHECK_MAX_AGE_SECONDS,
     PREPROD_ARTIFACT_EVENTS_TOPIC,
 )
+from launchpad.server import get_server_config
 from launchpad.utils.arroyo_metrics import DatadogMetricsBackend
-from launchpad.utils.logging import get_logger
+from launchpad.utils.logging import get_logger, setup_logging
 
 logger = get_logger(__name__)
 
 # Schema codec for preprod artifact events
 PREPROD_ARTIFACT_SCHEMA = get_codec(PREPROD_ARTIFACT_EVENTS_TOPIC)
 
 
+def process_kafka_message_with_service(msg: Message[KafkaPayload]) -> Any:
+    """Process a Kafka message using the actual service logic in a worker process."""
+
+    def worker_signal_handler(signum: int, frame) -> None:
+        logger.info(f"Worker process {os.getpid()} received signal {signum}, exiting...")
+        os._exit(0)
+
+    signal.signal(signal.SIGTERM, worker_signal_handler)
+    signal.signal(signal.SIGINT, worker_signal_handler)
+
+    if not logging.getLogger().handlers:
+        server_config = get_server_config()
+        setup_logging(verbose=server_config.debug, quiet=not server_config.debug)
+
+    try:
+        decoded = PREPROD_ARTIFACT_SCHEMA.decode(msg.payload.value)
+        ArtifactProcessor.process_message(decoded)
+        return decoded  # type: ignore[no-any-return]
+    except Exception as e:
+        logger.error(f"Failed to process message in worker: {e}", exc_info=True)
```

Comment:
I learned recently that `logger.error(..., exc_info=True)` is the same as `logger.exception(...)` so we should use the latter in these situations.